parsers/endorphin-php-4/scripts/parse.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+declare(strict_types = 1);
+ini_set('memory_limit', '-1');
+ini_set('max_execution_time', '-1');
+
+$benchmarkPos = array_search('--benchmark', $argv);
+$benchmark    = false;
+
+if ($benchmarkPos !== false) {
+    $benchmark = true;
+    unset($argv[$benchmarkPos]);
+    $argv = array_values($argv);
+}
+
+$agentListFile = $argv[1];
+
+$results   = [];
+$parseTime = 0;
+
+$start = microtime(true);
+require_once __DIR__ . '/../vendor/autoload.php';
+
+use EndorphinStudio\Detector\Detector;
+
+$detector = new Detector();
+
+$detector->analyse('Test String');
```

Comment:
Why is the detector analysing "test string" here?

Reply:
It's done to "warm up" the parser, mainly for when it's benchmarked (this is done in all of the parsers).